code/API_definitions/qod-api.yaml

```diff
@@ -433,6 +476,25 @@ components:
             - expiresAt
             - qosStatus
 
+    SessionInfoList:
+      description: Paged response of active sessions created by the user
+      type: object
+      required:
+        - perPage
+        - page
```

Comment:
perPage and page shouldn't be mandatory because this type is also used for "whole list" return with status 200

Reply:
What about situation, when user defines in example page = 10 and there are not so many active sessions, so the response should be 206 and sessions = empty array or 400? How should we handle such cases?

Reply:
To be honest I would leave this perPage and page parameters every time, even if whole list is returned. I would also add another parameter - "totalPages" so user can know how many pages are available for given perPage value

Reply:
I think if the server returns the whole list it should be 200.
With the pagination the client just says, if there are more than x records, please don't send me all at once.

Reply:
But if the client request doesn't contain page params, what should pagesize should we assume?

Reply:
I don't think we need totalPages. The client can just repeat the request with higher pages as long as he receives 206, and stop at 200.

Reply:
I would reference here to HTTP 206 in the Guideline: 

> 206: the response does not include the complete list of resources

So I would per default return maximum 20 entries per page.
IF the requester does not request with the page-parameters
AND the entries are more than 21
THEN return a list of 20 entries with a 206 response.

Reply:
Yeah, if server returns the whole list it should be 200 for sure. 
About pagination properties - if client does not specify them we should use the default values. In the issue was mentioned 100 as a default perPage value, but maybe it is too large, so 20 can stay.

Reply:
This is an implementation issue. I'm not sure, if we need for that a consensus on Camara level. But we can discuss about it in the community.

Reply:
But again, what about situation when user give the page value more than the last available page? 
So let's say that there are 3 pages but user request page=4 -> the response should be 400 Bad Request?
That's why I wanted to add totalPages in the response so user can know it

Reply:
> Yeah, if server returns the whole list it should be 200 for sure. About pagination properties - if client does not specify them we should use the default values. In the issue was mentioned 100 as a default perPage value, but maybe it is too large, so 20 can stay.

I don't think that we should response anything with pages, if the client didn't request with pages

Reply:
> But again, what about situation when user give the page value more than the last available page? So let's say that there are 3 pages but user request page=4 -> the response should be 400 Bad Request? That's why I wanted to add totalPages in the response so user can know it

Yes, that's not so easy... Here actually I could imagine a 404. :-) 
400 would mean that the parameter is generally wrong. But it isn't, we just don't have those resources.

Reply:
I'm expecting some discussions in the Camara group :-)

Reply:
> > Yeah, if server returns the whole list it should be 200 for sure. About pagination properties - if client does not specify them we should use the default values. In the issue was mentioned 100 as a default perPage value, but maybe it is too large, so 20 can stay.
> 
> I don't think that we should response anything with pages, if the client didn't request with pages

I mean for few records. For many it could be possible. But it is also an implementation issue, so we don't have to decide it for the Swagger file.

Reply:
@kacper-kicinski about your specific case when a page is "out-of-bound":
- It's no 400 Bad request -> The request was actually in a good syntax. The requester cannot know how many pages are there.
- It's more a 404 Not Found 
-> if the page != 1 -> 404

Reply:
> > But again, what about situation when user give the page value more than the last available page? So let's say that there are 3 pages but user request page=4 -> the response should be 400 Bad Request? That's why I wanted to add totalPages in the response so user can know it
> 
> Yes, that's not so easy... Here actually I could imagine a 404. :-) 400 would mean that the parameter is generally wrong. But it isn't, we just don't have those resources.

But for me a 206 with an empty list could be also possible.

Reply:
Then I would say @kacper-kicinski please add the 404 again, and then we will continue the discussion in the official PR.

Reply:
done

Reply:
So can I now merge and then create main PR to CAMARA?

Reply:
You need a feature branch in your forked repo and then create a PR from that to Camara QoD main.